src/fractal_faim_hcs/MetaSeriesUtils_dask.py

```diff
@@ -0,0 +1,302 @@
+            (1,) * len(fns_da.shape) + (3,), (*fns_da.shape, 3)
+        ),
+        new_axis=len(fns_da.shape),
+        meta=np.asanyarray([]).astype("float64"),
+    )
+
+    z_positions_full_planes = da.mean(
+        stage_positions_da[:,channels_full_planes,:,2],
+        axis=0
+    ).compute()
+    z_positions_one_plane = da.mean(
+        stage_positions_da[:,channels_one_plane,0,2],
+        axis=0
+    ).compute()
+
+    z_sampling = compute_z_sampling(z_positions_full_planes)
+
+    # roll single-plane channels to correct position
+    min_z = z_positions_full_planes.min(axis = 1).mean()
+    max_z = z_positions_full_planes.max(axis = 1).mean()
+    step_size = (max_z - min_z) / len(z_positions_full_planes[0])
+    for c, z_positions in zip(channels_one_plane, z_positions_one_plane):
+        shift_z = int((z_positions - min_z) // step_size)
+        fns_np[:,c,:] = np.roll(fns_np[:,c,:], shift_z, axis=1)
+
+
+    # MONTAGE PLANES
+    def _fuse_xy(
+        x: ArrayLike,
+        img_shape: tuple[int] = None,
+        img_dtype: np.dtype = None
+    ) -> ArrayLike:
+        fns = x.flatten()
+        if '' not in fns:
+            # TODO: Rewrite get_img_YX to reduce unnecessary computations
+            files = pd.DataFrame(fns, columns=["path"])
+            _, img, *_ = get_img_YX(assemble_fn, files)
+        else:
+            # return zeros if plane is empty
+            if img_shape != None and img_dtype != None:
+                img = np.zeros(img_shape, dtype=img_dtype)
+            else:
+                raise RuntimeError("'img_dimension' and 'img_dtype' must be provided if plane is empty")
+        return img.reshape(x.shape[1:] + img.shape)
+
+    # calculate one plane to get dimensions and dtype
+    # (choose a chanel which is non-empyt)
+    # TODO: could be done without loading images
+    sample_fused = _fuse_xy(fns_np[:, channels_full_planes[0], 0])
+    (nx_tot, ny_tot) = sample_fused.shape
+    img_dtype = sample_fused.dtype
+
+    # create dask array of assembled planes
+    fns_da = da.from_array(
+        fns_np,
+        chunks=((fns_np.shape[0],)+(1,)*len(fns_np.shape[1:]))
+    )
+    imgs_fused_da = da.map_blocks(
+        _fuse_xy,
+        fns_da,
+        chunks=da.core.normalize_chunks(
+            (1,) * len(fns_da.shape[1:]) + (nx_tot, ny_tot),
+            shape=fns_da.shape[1:] + (nx_tot, ny_tot),
+        ),
+        drop_axis=0,
+        new_axis=range(len(fns_da.shape) - 1, len(fns_da.shape) + 1),
+        meta=np.asanyarray([]).astype(sample_fused.dtype),
+    
+        img_shape=(nx_tot, ny_tot),
+        img_dtype=img_dtype
+    )
+
+
+    # GENERATE ROI_TABLES AND PX_METADATA:
+    # (by only loading one plane)
+    # TODO: Could also be done without loading images of entire plane
+    fns_sel = fns_da[:, channels_full_planes[0], 0].compute()
+    files = pd.DataFrame(fns_sel, columns=["path"])
+    px_metadata, _, _, _, roi_tables = get_img_YX(assemble_fn, files)
```

Comment:
We may need to refactor what happening here. Weird if we call the `get_img_YX` function only to get metadata and roi_tables. Not really what that function was made for